rt/rs/description-swagger-ui/src/main/java/org/apache/cxf/jaxrs/swagger/ui/OsgiSwaggerUiResolver.java

```diff
@@ -32,18 +31,24 @@ public class OsgiSwaggerUiResolver extends SwaggerUiResolver {
     };
     
     private final Class<? extends Annotation> annotationBundle;
+    private FrameworkUtilWrapper frameworkUtilWrapper;
```

Comment:
I believe the the introduction of the wrapper is for testability reasons? I would suggest to go with integration tests instead, it would also help to catch any changes in Karaf (I could help with that).

Reply:
Thanks for the review @reta. Yes, the wrapper is for testability as FrameworkUtil is static. But I agree pax exam tests are a better idea than mocking OSGi. I added some but I couldn't get them to work on the main branch, due to some dependencies missing such as spring 6.0.0-M5 (I couldn't find it in central maven repo either - not sure how I can fetch these). So I opened a new PR based on 3.5.x-fixes: https://github.com/apache/cxf/pull/982

Reply:
Thank you @awrb , yes, `main` is in transition to Jakarta 9.1+ support, we could close this one (I will port the fix from 3.5.x to all other branches), thanks again!